stan/math/prim/mat/err/check_pos_semidefinite.hpp

```diff
@@ -42,13 +42,36 @@ inline void check_pos_semidefinite(
   using Eigen::Dynamic;
   using Eigen::LDLT;
   using Eigen::Matrix;
-  LDLT<Matrix<double, Dynamic, Dynamic> > cholesky = value_of_rec(y).ldlt();
+  LDLT<Matrix<double, Dynamic, Dynamic>> cholesky = value_of_rec(y).ldlt();
   if (cholesky.info() != Eigen::Success
       || (cholesky.vectorD().array() < 0.0).any())
     domain_error(function, name, "is not positive semi-definite.", "");
   check_not_nan(function, name, y);
 }
 
+/**
+ * Check if the specified matrix is positive semidefinite
+ *
+ * @tparam Derived Derived type of the Eigen::LDLT transform.
+ * @param function Function name (for error messages)
+ * @param name Variable name (for error messages)
+ * @param cholesky Eigen::LDLT to test, whose progenitor matrix must
+ * not have any NaN elements.
+ * @throw <code>std::domain_error</code> if the matrix is not positive
+ *   semi-definite.
+ */
+template <typename Derived>
+inline void check_pos_semidefinite(const char *function, const char *name,
+                                   const Eigen::LDLT<Derived> &cholesky) {
+  // From the Eigen::LDLT we cannot check for NaNs in the original
```

Comment:
Could you move this documentation to the doxygen comment? To be concrete, I think that you can omit the fact that we don't check NaNs (I believe we've already checked that to get an `Eigen::LDLT` object), we can remove the symmetry comment also.

What should be included is when describing the variable `cholesky` it should mention that it's requiring that the size is greater than 0. (Is that true? Can we not construct an `Eigen::LDLT` with a 0x0 matrix?)

Reply:
The Eigen docs don't specify. But

```
Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> y2(0, 0);
Eigen::LDLT<Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> > y_ldlt2(y2);
```

is enough to cause a segfault on my machine.

We can't make an Eigen::LDLT out of a rectangular matrix, Eigen throws an exception. But Eigen::LDLT will happily consume a non-symmetric square matrix as if it were symmetric and only use half of it. We can even hide nans in the other half and not get errors. (There's a test in the PR demonstrating this.)

I guess just delete the whole comment. the LDLT object passed in needs to be valid and its fair to leave that assumption unstated.